File: maths-house/TEMP_MAUI/BL/QuestionResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GCMS.MathHouse
{
    public class QuestionResult
    {
        public string QuestionText { get; set; }
        public int CorrectAnswer { get; set; }
        public int UserAnswer { get; set; }
        public bool IsCorrect => UserAnswer == CorrectAnswer;
    }

}

File: maths-house/src/app/core/models/question-result.ts
export interface QuestionResult {
  questionText: string;
  correctAnswer: number;
  userAnswer?: number;
  get isCorrect(): boolean;
}

export function makeResult(questionText: string, correctAnswer: number): QuestionResult {
  return {
    questionText,
    correctAnswer,
    get isCorrect() { return this.userAnswer === this.correctAnswer; }
  } as QuestionResult;
}